internal/awsnuke/client_integration_test.go

```diff
@@ -0,0 +1,89 @@
+package awsnuke_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/aws-nuke/internal/awsnuke"
+)
+
+var AWSAccountId = "353720269506"
+
+func TestDestroyEC2Instances(t *testing.T) {
+	createNewInstance(t, "us-east-1", "ami-6057e21a")
+
+	nuke := awsnuke.New(AWSAccountId)
+	instances, err := nuke.ListNonProtectedEC2Instances()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	err = nuke.DestroyEC2Instances(instances)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	instances, err = nuke.ListNonProtectedEC2Instances()
```

Comment:
It's a bit odd to use the library to check for instances, as the library itself may be broken. Moreover, this will fail if this test is running in parallel with other tests that are creating EC2 instances.

Is there some other check you could do? For example, [Terratest](https://github.com/gruntwork-io/terratest) has a function for a short unique ID you can generate and add as a tag to each instance. You can then search for that tag and make sure you can no longer find it.

Reply:
This was a quick fix to get it working. I agree that the assertion should not rely on the Unit Under Test. Will fix.